🐛  warning: 'TARGET_OS_MAC' is not defined, evaluates to 0 [-Wundef-prefix=TARGET_OS_]

**Fiber version**
v2.20.1

**Issue description**
If I use monitor middleware and try to build my project on macOS using [LLVM/Clang 11.0](https://en.wikipedia.org/wiki/Xcode#Xcode_11.x_-_13.x_(since_SwiftUI_framework)_2) or higher (in my case 12.0.0), I get the following error:
```
# github.com/gofiber/fiber/v2/internal/gopsutil/cpu
../../Go/pkg/mod/github.com/gofiber/fiber/v2@v2.20.1/internal/gopsutil/cpu/cpu_darwin_cgo.go:13:5: warning: 'TARGET_OS_MAC' is not defined, evaluates to 0 [-Wundef-prefix=TARGET_OS_]
```

**Code snippet**

```go
package main

import (
	"log"
	"github.com/gofiber/fiber/v2"
	"github.com/gofiber/fiber/v2/middleware/monitor"
)

func main() {
	app := fiber.New()
	
	app.Get("/dashboard", monitor.New())
	
	log.Fatal(app.Listen(":3000"))
}

```

**Proposed Solution**
Based on this [issue](https://github.com/shirou/gopsutil/issues/976) it seems adding `#include <TargetConditionals.h>` before using `TARGET_OS_MAC` solves the problem.
As a workaround calling `CGO_CFLAGS=-Wno-undef-prefix go build` remove the warning message.

--- a/internal/gopsutil/cpu/cpu_darwin_cgo.go
+++ b/internal/gopsutil/cpu/cpu_darwin_cgo.go
@@ -10,6 +10,7 @@ package cpu
 #include <mach/mach_init.h>
 #include <mach/mach_host.h>
 #include <mach/host_info.h>
+#include <TargetConditionals.h>
 #if TARGET_OS_MAC
 #include <libproc.h>
 #endif

